Automattic/wp-calypso
# Settings: Allow Users to Insert Header Code

Fixes #56023

Requires Automattic/jetpack#34881 (also, I need to add a version check here once that Jetpack PR is merged)

## Proposed Changes

Allows user to directly edit their `<head>` element without relying on a third-party plugin. A plugin to insert code in the Header element is now the eighth most popular on WP.com (!!), so hopefully this will make things easier for both users and those in Support. 

Rather than using a default textbox which created a strange experience, it felt more appropriate to use CodeMirror. Unfortunately, that means adding another package, but I've used the same one already used in Automattic projects like Newspack so I don't think that's a problem. 

## Testing Instructions

1. Apply the Jetpack PR
2. Go to Settings > General and scroll to "Site Tools" on an Atomic or Jetpack site (this didn't really seem to fit within any section, and the section cards are quite small for a code editor!)
3. Select "Header code"
4. Enter some header code, and verify that it's included on your site 

<img width="765" alt="Screenshot 2024-01-08 at 08 30 22" src="https://github.com/Automattic/wp-calypso/assets/43215253/84a9d139-5482-405b-8bf1-72d308a6b634">

## Pre-merge Checklist

Can't access P2s, but have tested on Simple and Atomic - the former can't access this page from Settings, and a direct link shows an upgrade nudge. 

- [] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?

cc @tyxla - it looks like you've worked on Settings regularly before, might this maybe fall into your remit? thanks!! :)
